addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/KNXBasicThingHandler.java

```
@@ -0,0 +1,682 @@
+                    if (!filledDescription && config.getFetch()) {
+                        if (descriptionJob == null || descriptionJob.isCancelled()) {
+                            descriptionJob = knxScheduler.schedule(descriptionRunnable, 0, TimeUnit.MILLISECONDS);
+                        }
+                    }
+                } else {
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+            }
+        } catch (KNXException e) {
+            logger.error("An error occurred while testing the reachability of a thing '{}' : {}", getThing().getUID(),
+                    e.getLocalizedMessage());
+            if (logger.isDebugEnabled()) {
+                logger.error("", e);
+            }
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getLocalizedMessage());
+        }
+    }
+
+    private Runnable descriptionRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            try {
+                if (getBridge().getStatus() == ThingStatus.ONLINE) {
+                    logger.debug("Fetching device information for address {}", address.toString());
+
+                    Thread.sleep(OPERATION_INTERVAL);
+                    byte[] data = getBridgeHandler().readDeviceDescription(address, 0, false, OPERATION_TIMEOUT);
+
+                    if (data != null) {
+                        final DD0 dd = DeviceDescriptor.DD0.fromType0(data);
+
+                        Map<String, String> properties = editProperties();
+                        properties.put(FIRMWARE_TYPE, Firmware.getName(dd.getFirmwareType()));
+                        properties.put(FIRMWARE_VERSION, Firmware.getName(dd.getFirmwareVersion()));
+                        properties.put(FIRMWARE_SUBVERSION, Firmware.getName(dd.getSubcode()));
+                        try {
+                            updateProperties(properties);
+                        } catch (Exception e) {
```

Do we really to catch and ignore all exceptions not some specif exception... 😢